lib/operations/create_indexes.js

```diff
@@ -2,60 +2,102 @@
 
 const Aspect = require('./operation').Aspect;
 const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
-const executeCommand = require('./db_ops').executeCommand;
+const CommandOperationV2 = require('./command_v2');
 const MongoError = require('../core').MongoError;
-const ReadPreference = require('../core').ReadPreference;
+const serverLacksFeature = require('../utils').serverLacksFeature;
+const parseIndexOptions = require('../utils').parseIndexOptions;
 
-class CreateIndexesOperation extends OperationBase {
-  constructor(collection, indexSpecs, options) {
-    super(options);
+const validIndexOptions = new Set([
+  'unique',
+  'partialFilterExpression',
+  'sparse',
+  'background',
+  'expireAfterSeconds',
+  'storageEngine',
+  'collation'
+]);
 
+class CreateIndexesOperation extends CommandOperationV2 {
+  /**
+   * @ignore
+   */
+  constructor(parent, collection, indexes, options) {
+    super(parent, options);
     this.collection = collection;
-    this.indexSpecs = indexSpecs;
+    this.indexes = indexes;
   }
 
-  execute(callback) {
-    const coll = this.collection;
-    const indexSpecs = this.indexSpecs;
-    let options = this.options;
+  /**
+   * @ignore
+   */
+  execute(server, callback) {
+    // by setting singular to true, ensure createIndex returns the name
+    // of the created index instead of the full result from server
+    // TODO: remove in v4 (breaking change)
```

Comment:
Couldn't figure out a way to avoid using a `singular` variable to determine what to return, but added a TODO to remove in v4.

Reply:
ah okay, well in that case I hate to be nitpicky, but I think you should convert the indexes in the constructor and use a descriptive property there (maybe `returnNameOnly`). What I meant with that other comment about matching the server is that I should generally be able to construct an operation and when printed it out it should look like the command manual form (this.collection and this.indexes should be those types described in the documentation).

as a rule of thumb I think its better that anything that does not require wire protocol version information be relegated to the constructor.

Reply:
That should be good now, lmk